execution_engine_testing/test_support/src/wasm_test_builder.rs

```diff
@@ -362,6 +379,16 @@ impl LmdbWasmTestBuilder {
         }
         self
     }
+
+    /// Returns a handle to the underlying LMDB environment.
+    pub fn lmdb_environment(&self) -> Arc<LmdbEnvironment> {
+        self.engine_state.lmdb_environment()
+    }
+
+    /// Returns a handle to the underlying LMDB database.
+    pub fn lmdb_handle(&self) -> Database {
+        self.engine_state.lmdb_handle()
+    }
```

Comment:
These two also leak the implementation detail. From what it looks like later on the usage side (`disk_use`) the only reason we need those is to call `cursor.iter()`. How about instead this, we add a method `iter()` and return an iterator over elements in the DB? That should be implementable for all types of DB backends:
* LMDB
* in-memory
* RocksDB